### PR TITLE
Add ability to perform or not scenarios depending on the previous res…

### PR DESCRIPTION
**Pull request by Yauhenda**
…ults

### CODE REVIEW

File: jbehave-core/src/main/java/org/jbehave/core/embedder/PerformableTree.java [hunk 1]
```diff
@@ -973,18 +983,48 @@ public void reportFailures(FailureContext context) {
...
+            }
+        }
+
+        private Optional<UUIDExceptionWrapper> causeToSkip(PerformableScenario scenario) {
+            Meta meta = scenario.getScenario().getMeta();
+            if (meta.hasProperty(DEPENDS_ON_ID)) {
+                String dependsOnId = meta.getProperty(DEPENDS_ON_ID);
+                Boolean performDependentScenario = successfulScenariosId.get(dependsOnId);
+                if (performDependentScenario == null) {
+                    return wrap(new DependsOnNotFoundException(SCENARIO_WITH_ID + dependsOnId
```
Comment (Yauhenda) [inline jbehave-core/src/main/java/org/jbehave/core/embedder/PerformableTree.java]: Changed messages

File: jbehave-core/src/main/java/org/jbehave/core/embedder/PerformableTree.java [hunk 2]
```diff
@@ -1171,19 +1219,29 @@ protected void performRestartableSteps(RunContext context)
...
+            perform(context, stories, Optional.empty());
+        }
+
+        void perform(RunContext context, GivenStories stories, Optional<UUIDExceptionWrapper> causeToSkip)
+                throws InterruptedException {
             perform(beforeSteps, context, r -> r.beforeScenarioSteps(Stage.BEFORE),
-                r -> r.afterScenarioSteps(Stage.BEFORE));
-            performGivenStories(context, givenStories, stories);
-            performRestartableSteps(context);
+                r -> r.afterScenarioSteps(Stage.BEFORE), Optional.empty());
```
Comment (Yauhenda) [inline jbehave-core/src/main/java/org/jbehave/core/embedder/PerformableTree.java]: Before and after scenario steps not marked with causeToSkip

File: jbehave-core/src/main/java/org/jbehave/core/embedder/PerformableTree.java [hunk 3]
```diff
@@ -1311,7 +1381,10 @@ public void perform(RunContext context) throws InterruptedException {
 					context.interruptIfCancelled();
 					if (ignoring) {
 						reporter.ignorable(step.asString(keywords));
-					} else {
+					} else if(causeToSkip.isPresent()) {
+                        reporter.beforeStep(step.asString(keywords));
```
Comment (Yauhenda) [inline jbehave-core/src/main/java/org/jbehave/core/embedder/PerformableTree.java]: Added obvious call of `beforeStep(String step)` method.

File: jbehave-core/src/main/java/org/jbehave/core/embedder/PerformableTree.java [hunk 4]
```diff
@@ -973,18 +983,48 @@ public void reportFailures(FailureContext context) {
...
+            if (meta.hasProperty(DEPENDS_ON_ID)) {
+                String dependsOnId = meta.getProperty(DEPENDS_ON_ID);
+                Boolean performDependentScenario = successfulScenariosId.get(dependsOnId);
+                if (performDependentScenario == null) {
+                    return wrap(new DependsOnNotFoundException(SCENARIO_WITH_ID + dependsOnId
+                            + " was not performed"));
+                }
+                if (!performDependentScenario) {
+                    return wrap(new DependsOnFailedException(SCENARIO_WITH_ID + dependsOnId
+                            + " was failed"));
```
Comment (valfirst) [inline jbehave-core/src/main/java/org/jbehave/core/embedder/PerformableTree.java]: "Scenario with @scenarioId " + dependsOnId +" upon which current one depends was failed"

File: jbehave-core/src/main/java/org/jbehave/core/embedder/PerformableTree.java [hunk 5]
```diff
@@ -973,18 +983,48 @@ public void reportFailures(FailureContext context) {
...
+        }
+
+        private Optional<UUIDExceptionWrapper> causeToSkip(PerformableScenario scenario) {
+            Meta meta = scenario.getScenario().getMeta();
+            if (meta.hasProperty(DEPENDS_ON_ID)) {
+                String dependsOnId = meta.getProperty(DEPENDS_ON_ID);
+                Boolean performDependentScenario = successfulScenariosId.get(dependsOnId);
+                if (performDependentScenario == null) {
+                    return wrap(new DependsOnNotFoundException(SCENARIO_WITH_ID + dependsOnId
+                            + " was not performed"));
```
Comment (valfirst) [inline jbehave-core/src/main/java/org/jbehave/core/embedder/PerformableTree.java]: "Scenario with @scenarioId " + dependsOnId +" upon which current one depends is not found"

File: jbehave-core/src/main/java/org/jbehave/core/embedder/PerformableTree.java (hunk 2 above)
Comment (valfirst) [reply jbehave-core/src/main/java/org/jbehave/core/embedder/PerformableTree.java]: what happens with After scenario steps declared in Lifecycle?

**Pull request closed without merging**